store/v2/transient/store.go

```diff
@@ -0,0 +1,46 @@
+package transient
+
+import (
+	dbm "github.com/cosmos/cosmos-sdk/db"
+	"github.com/cosmos/cosmos-sdk/db/memdb"
+	"github.com/cosmos/cosmos-sdk/store/types"
+	"github.com/cosmos/cosmos-sdk/store/v2/dbadapter"
+)
+
+var (
+	_ types.KVStore   = (*Store)(nil)
+	_ types.Committer = (*Store)(nil)
+)
+
+// Store is a wrapper for a memory store which does not persist data.
+type Store struct {
+	dbadapter.Store
+	conn dbm.DBConnection
+}
+
+// NewStore constructs a new transient store.
+func NewStore() *Store {
+	db := memdb.NewDB()
+	return &Store{
+		Store: dbadapter.Store{DB: db.ReadWriter()},
+		conn:  db,
+	}
+}
+
+// Implements Store.
+func (ts *Store) GetStoreType() types.StoreType {
+	return types.StoreTypeTransient
+}
+
+// Implements CommitStore
+// Commit cleans up Store.
+func (ts *Store) Commit() (id types.CommitID) {
+	ts.DB.Discard()
+	ts.Store = dbadapter.Store{DB: ts.conn.ReadWriter()}
+	return
+}
+
+func (ts *Store) SetPruning(types.PruningOptions)  {}
```

Comment:
Is it worth logging something here, letting the caller know they have invoked a noop?

Reply:
Probably not a big deal - this is copied from the [old version](https://github.com/cosmos/cosmos-sdk/blob/master/store/transient/store.go) and shouldn't actually be called by anything. it's an extraneous part of the `Committer` store interface.

However, now that I think of it, we could probably remove the `v2/mem` and `v2/transient` stores altogether. They are only used as part of the root store now and just wrap the `dbadapter.Store`